### PR TITLE
feat: add `lwc:spread` directive

### PR DESCRIPTION
**Pull request by abdulsattar**
## Details
RFC: https://github.com/salesforce/lwc-rfcs/pull/52
Adds `lwc:spread` directive that accepts properties and assigns them to the element.

## Does this pull request introduce a breaking change?

* ✅ No, it does not introduce a breaking change.

## Does this pull request introduce an observable change?

* ✅ No, it does not introduce an observable change.

### CODE REVIEW

**Comment by abdulsattar** [comment]
`lwc-platform` tests are failing because of metadata mismatch. I'll fix them when we merged this and upgrade lwc version in lwc-platform.

**Comment by nolanlawson** [comment]
/nucleus ignore --reason "lwc-platform tests are failing because of metadata mismatch"

File: packages/@lwc/errors/src/compiler/error-info/template-transform.ts [hunk 1]
```diff
@@ -633,4 +633,17 @@ export const ParserDiagnostics = {
...
+            'Invalid lwc:spread usage. The `lwc:spread` Directive must be enabled in order to use this feature.',
+        level: DiagnosticLevel.Error,
+        url: '',
+    },
+    INVALID_LWC_SPREAD_LITERAL_PROP: {
+        code: 1155,
+        message:
+            'Invalid lwc:spread usage on element "{0}". The directive binding must be an expression.',
+        level: DiagnosticLevel.Error,
+    },
```
Comment (nolanlawson) [inline packages/@lwc/errors/src/compiler/error-info/template-transform.ts:648]: What happens if `lwc:spread` is used on a `<template>`? E.g. a `<template for:each>`?

File: packages/@lwc/integration-karma/test/spread/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    });
+
+    it('should assign props to standard elements', () => {
+        expect(elm.shadowRoot.querySelector('span').className).toEqual('spanclass');
+    });
+    it('should assign props to dynamic elements', () => {
+        expect(
+            elm.shadowRoot.querySelector('x-cmp').shadowRoot.querySelector('span').textContent
+        ).toEqual('Name: Dynamic');
+    });
```
Comment (nolanlawson) [inline packages/@lwc/integration-karma/test/spread/index.spec.js:42]: Some more test cases that may be worth testing:

- The key inside of the `spanProps` is changed
- The `className` key in `spanProps` is deleted
- The value of `className` in `spanProps` is set to `undefined`
- The entire `spanProps` object changes to a new object like `{ className: 'foo' }`
- The entire `spanProps` object changes to an empty object

Comment (abdulsattar) [reply packages/@lwc/integration-karma/test/spread/index.spec.js:42]: I completely missed the reactivity part of it. I actually found a few issues with reactivity after adding these tests.

**Comment by abdulsattar** [comment]
/nucleus ignore --reason "lwc-platform tests are failing because of metadata mismatch"

**Pull request merged**